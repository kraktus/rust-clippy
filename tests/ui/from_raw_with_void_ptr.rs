#![warn(clippy::from_raw_with_void_ptr)]

use std::ffi::c_void;

fn main() {
    // must lint
    let ptr = Box::into_raw(Box::new(42usize)) as *mut c_void;
    let _ = unsafe { Box::from_raw(ptr) };

    // shouldn't be linted
    let _ = unsafe { Box::from_raw(ptr as *mut usize) };

    // shouldn't be linted
    let should_not_lint_ptr = Box::into_raw(Box::new(12u8)) as *mut u8;
    let _ = unsafe { Box::from_raw(should_not_lint_ptr as *mut u8) };
}
