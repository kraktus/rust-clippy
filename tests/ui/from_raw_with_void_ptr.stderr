error: creating a `Box` from a raw void pointer
  --> $DIR/from_raw_with_void_ptr.rs:8:22
   |
LL |     let _ = unsafe { Box::from_raw(ptr) };
   |                      ^^^^^^^^^^^^^^^^^^
   |
help: cast this to a pointer of the actual type
  --> $DIR/from_raw_with_void_ptr.rs:8:36
   |
LL |     let _ = unsafe { Box::from_raw(ptr) };
   |                                    ^^^
   = note: `-D clippy::from-raw-with-void-ptr` implied by `-D warnings`

error: aborting due to previous error

